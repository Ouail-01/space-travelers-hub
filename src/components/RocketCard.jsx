import { PropTypes } from 'prop-types';

function RocketCard({ rocket }) {
  return (
    <li className="rocket-card">
      <img src={rocket.flickr_images[0]} alt="Rocket" />
      <div className="rocket-card-details">
        <h2 className="rocket-card-title">{rocket.name}</h2>
        <p className="rocket-card-description">{rocket.description}</p>
        <button type="button" className="reserve-btn">Reserve Rocket</button>
      </div>
    </li>
  );
}

RocketCard.propTypes = {
  rocket: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    flickr_images: PropTypes.arrayOf(PropTypes.string).isRequired,
  }).isRequired,
};

export default RocketCard;
