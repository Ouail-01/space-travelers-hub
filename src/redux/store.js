import { configureStore } from '@reduxjs/toolkit';
import rocketsSlice from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsSlice,
    // categories: categoryReducer,
  },
});

export default store;
