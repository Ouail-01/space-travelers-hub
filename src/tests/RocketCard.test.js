import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import RocketCard from '../components/RocketCard';
import store from '../redux/store';

describe('Test component\'s correct rendering', () => {
  test('test site\'s rendering', () => {
    const rocket = {
      id: 'item_1',
      name: 'Falcon Heavy',
      description: 'Falcon Heavy Description',
      flickr_images: [
        'https://farm5.staticflickr.com/4599/38583829295_581f34dd84_b.jpg',
        'https://farm5.staticflickr.com/4645/38583830575_3f0f7215e6_b.jpg',
      ],
    };
    render(
      <Provider store={store}>
        <RocketCard key={rocket.id} rocket={rocket} />
      </Provider>,
    );
    const rocketName = screen.queryByRole('heading', { level: 2 });
    expect(rocketName.innerHTML).toBe('Falcon Heavy');
  });
});

describe('test conditional elemets\' display', () => {
  test('Test display of \'Reserve Rocket\'', () => {
    const rocket = {
      id: 'item_1',
      name: 'Falcon Heavy',
      description: 'Falcon Heavy Description',
      flickr_images: [
        'https://farm5.staticflickr.com/4599/38583829295_581f34dd84_b.jpg',
        'https://farm5.staticflickr.com/4645/38583830575_3f0f7215e6_b.jpg',
      ],
      reserved: false,
    };
    render(
      <Provider store={store}>
        <RocketCard key={rocket.id} rocket={rocket} />
      </Provider>,
    );
    const reserveButton = screen.queryByRole('button');
    expect(reserveButton.innerHTML).toBe('Reserve Rocket');
  });

  test('Test display of \'Cancel Reservation\'', () => {
    const rocket = {
      id: 'item_1',
      name: 'Falcon Heavy',
      description: 'Falcon Heavy Description',
      flickr_images: [
        'https://farm5.staticflickr.com/4599/38583829295_581f34dd84_b.jpg',
        'https://farm5.staticflickr.com/4645/38583830575_3f0f7215e6_b.jpg',
      ],
      reserved: true,
    };
    render(
      <Provider store={store}>
        <RocketCard key={rocket.id} rocket={rocket} />
      </Provider>,
    );
    const reserveButton = screen.queryByRole('button');
    expect(reserveButton.innerHTML).toBe('Cancel Reservation');
  });

  test('Test display of \'Reserved\' badge', () => {
    const rocket = {
      id: 'item_1',
      name: 'Falcon Heavy',
      description: 'Falcon Heavy Description',
      flickr_images: [
        'https://farm5.staticflickr.com/4599/38583829295_581f34dd84_b.jpg',
        'https://farm5.staticflickr.com/4645/38583830575_3f0f7215e6_b.jpg',
      ],
      reserved: true,
    };
    render(
      <Provider store={store}>
        <RocketCard key={rocket.id} rocket={rocket} />
      </Provider>,
    );
    const reserveBadge = screen.queryByText('Reserved');
    expect(reserveBadge).toBeInTheDocument();
  });

  test('Test not displaying \'Reserved\' badge when not reserved', () => {
    const rocket = {
      id: 'item_1',
      name: 'Falcon Heavy',
      description: 'Falcon Heavy Description',
      flickr_images: [
        'https://farm5.staticflickr.com/4599/38583829295_581f34dd84_b.jpg',
        'https://farm5.staticflickr.com/4645/38583830575_3f0f7215e6_b.jpg',
      ],
      reserved: false,
    };
    render(
      <Provider store={store}>
        <RocketCard key={rocket.id} rocket={rocket} />
      </Provider>,
    );
    const reserveBadge = screen.queryByText('Reserved');
    expect(reserveBadge).not.toBeInTheDocument();
  });
});

describe('snapshot test', () => {
  it('page renders', () => {
    const rocket = {
      id: 'item_1',
      name: 'Falcon Heavy',
      description: 'Falcon Heavy Description',
      flickr_images: [
        'https://farm5.staticflickr.com/4599/38583829295_581f34dd84_b.jpg',
        'https://farm5.staticflickr.com/4645/38583830575_3f0f7215e6_b.jpg',
      ],
      reserved: true,
    };
    const rocketCard = renderer.create(
      <Provider store={store}>
        <RocketCard rocket={rocket} />
      </Provider>,
    );

    const tree = rocketCard.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
