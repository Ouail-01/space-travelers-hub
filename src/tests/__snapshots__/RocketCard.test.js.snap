// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test page renders 1`] = `
<li
  className="rocket-card"
>
  <img
    alt="Rocket"
    src="https://farm5.staticflickr.com/4599/38583829295_581f34dd84_b.jpg"
  />
  <div
    className="rocket-card-details"
  >
    <h2
      className="rocket-card-title"
      data-testid="rocket-header"
    >
      Falcon Heavy
    </h2>
    <p
      className="rocket-card-description"
    >
      <span
        className="rocket-reserve-badge"
        data-testid="reserved-badge"
      >
        Reserved
      </span>
      Falcon Heavy Description
    </p>
    <button
      className="cancel-btn"
      data-testid="cancel-button"
      onClick={[Function]}
      type="button"
    >
      Cancel Reservation
    </button>
  </div>
</li>
`;
